lib/rdoc/code_object/context.rb

```diff
@@ -1261,4 +1261,50 @@ def upgrade_to_class mod, class_type, enclosing
 
   autoload :Section, "#{__dir__}/context/section"
 
+  ##
+  # Attempts to locate the module object in this context.
+  #
+  # The scoping rules of Ruby to resolve the name of an included module are:
+  # - first search constant directly defined in nested modules from inside to outside
+  # - if not found, look into the children of included modules,
+  #   in reverse inclusion order;
+  # - if still not found, look into included modules of Object
+
+  def lookup_module(name, before: nil, searched: {}.compare_by_identity)
+    # Search nested modules first
+    nesting = self
+    while nesting
+      full_name = nesting.child_name(name)
+      mod = @store.modules_hash[full_name]
+      return mod if mod
+      nesting = nesting.parent
```

Comment:
I found that this part is not correct. should use Module.nesting instead of `CodeObject#parent` chain.
```diff
- while nesting
-   ...
-   nesting = nesting.parent
- end
+ unavailable_data.module_nesting.each do
+   ...
+ end
```

But I think RDoc::Include and RDoc::Extend does not provide enough information.
This pull request and the original implementation uses `CodeObject#parent` chain instead of module nesting.

Complicated example
```ruby
class ::A
  class ::B
    class ::A
      class ::B::C
        # Module.nesting is [::B::C, ::A, ::B, ::A]
        include X
      end
    end
  end
end
```